src/matrix/lkj.jl

```diff
@@ -186,14 +186,16 @@ end
 #  for 1 / c₀, even if they say that it is not.
 #  -----------------------------------------------------------------------------
 
-function lkj_onion_loginvconst(d::Integer, η::Real)
+function lkj_onion_loginvconst(d::Integer, η::T) where {T <: Real}
     #  Equation (17) in LKJ (2009 JMA)
-    sumlogs = zero(η)
-    for k in 2:d - 1
-        sumlogs += 0.5k*logπ + loggamma(η + 0.5(d - 1 - k))
+    h = convert(float(T), 0.5)
```

Comment:
There are still problems below with e.g. `(2η + d - 3)*logtwo` - that will be of type `Float64` if `η` is an integer. Hence some more changes are needed. It also seems we should take into account the type of `d` here - if it is eg. a `BigInt` then we would want to work with `BigFloat` since e.g. `::BigInt * ::Float64 -> ::BigFloat`.
Maybe we need to convert irrationals etc. to something like
```suggestion
    T = float(Base.promote_typeof(d, η))
    h = T(1//2)
```
(`T` in the function signature could be reverted to `Real`)